docs(readme): add api details, cli usage and more (#124)
godaddy/warehouse.ai: README.md

Scalable Object Ledger and CDN.

## Introducing Warehouse 7

Warehouse 7, also known as Warehouse Next, is a scalabe object ledger and cdn system for powering dynamic web applications. The new system has been entirly redisgned using a lightweight, portable and flexible architecture.

While designing the new system, it made sense to add a higher level of abstraction.
The idea is expose a generic Object API and a CDN service. The user can use the CDN service to upload the desired assets while using the Object API to store the metadata of the assets (e.g., file urls, etc.)

> The Object API acts as generic ledger and it can be used by a third-party system to store any type of data.

Reviewer: ```suggestion
The Object API acts as generic ledger and it can be used to store any type of data. In particular, the Object API is good at storing versioned JSON data that may have one or more "head" version(s) for one or more "environment(s)"; for example, "production" and "staging".
```

> Warehouse can be used for different use cases in different ways. At GoDaddy we mainly use it for storing and serving our web applications compiled assets. Our apps workflow can be summarized as following:

Reviewer: ```suggestion
Warehouse can be used for different use cases in different ways. At GoDaddy we mainly use it for storing and serving our web applications' compiled browser-side assets. Our applications' workflow can be summarized as follows:
```

1. Engineers create a new service or package version and release it (e.g., `npm publish`, `git push <tag>`)

> 1. A CICD workflow is triggered and runs the Webpack builds

Reviewer: ```suggestion
1. A CICD workflow is triggered and runs the builds (e.g., webpack)
```

> 1. When all builds are completed, the CICD workflow uploads the assets to Warehouse

Reviewer: ```suggestion
1. When all builds are completed, the CICD workflow uploads the assets to Warehouse using the CDN API
1. Metadata for those assets on the CDN are stored in the Object API to allow application servers to fetch the URL(s) for the latest assets for a given environment.
```

> ## Get Started

Reviewer: ```suggestion
## Get Started

Warehouse is a Node.js web service running on Fastify. It stores data in Amazon's DynamoDB.
```

### Quick Start

```bash
git clone git@github.com:godaddy/warehouse.ai.git
cd warehouse.ai && npm install

> npm start

Reviewer: Is there needed DynamoDB env config before starting?

Author: Nope, needs localstack only

```

### Programmatic Usage

> 

Reviewer: ```suggestion

If you need further customization of Warehouse such as authentication and/or custom logging, you may wrap it using a `fastify` application.

```
